app/controllers/declaration/DeclarantDetailsController.scala

```diff
@@ -43,28 +45,30 @@ class DeclarantDetailsController @Inject()(
 
   def displayPage(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
     request.cacheModel.parties.declarantDetails match {
-      case Some(data) => Ok(declarantDetailsPage(mode, form().fill(data)))
-      case _          => Ok(declarantDetailsPage(mode, form()))
+      case Some(_) => Ok(declarantDetailsPage(mode, form().fill(DeclarantEoirConfirmation(YesNoAnswers.yes))))
+      case _       => Ok(declarantDetailsPage(mode, form()))
     }
   }
 
-  def saveAddress(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
+  def submitForm(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
     form()
       .bindFromRequest()
       .fold(
         formWithErrors => Future.successful(BadRequest(declarantDetailsPage(mode, formWithErrors))),
-        validDeclarantDetails =>
-          updateCache(validDeclarantDetails)
-            .map(_ => navigator.continueTo(mode, controllers.declaration.routes.ExporterDetailsController.displayPage))
+        validForm =>
+          if (validForm.answer == YesNoAnswers.yes)
+            updateCache(DeclarantDetails(EntityDetails(Some(Eori(request.eori)), None)))
+              .map(_ => navigator.continueTo(mode, controllers.declaration.routes.ExporterDetailsController.displayPage))
+          else
+            Future.successful(SeeOther(controllers.declaration.routes.NotEligibleController.displayNotDeclarant().url).withNewSession)
```

Comment:
Why SeeOther and .url?

Is `Redirect(controllers.declaration.routes.NotEligibleController.displayNotDeclarant()).withNewSession`

Reply:
Thanks